bad_data/README.md

```diff
@@ -22,3 +22,5 @@ These are files used for reproducing various bugs that have been reported.
 
 * PARQUET-1481.parquet: tests a case where a schema Thrift value has been
   corrupted
+
+* GH_43605.parquet: In Go, file reader goroutine crashed.
```

Comment:
Naming isn't terrific as it doesn't say which GH repo this is in. Perhaps `ARROW-GH-43605.parquet`?

Reply:
I agree, it is done.